tests/test_utils.py

```diff
@@ -598,4 +601,53 @@ def test_should_bypass_proxies_no_proxy(
     """
     no_proxy = '192.168.0.0/24,127.0.0.1,localhost.localdomain,172.16.1.1'
     # Test 'no_proxy' argument
+    _proxy_bypass_cached.cache.clear()
     assert should_bypass_proxies(url, no_proxy=no_proxy) == expected
+
+
+@pytest.mark.skipif(os.name != 'nt', reason='Test only on Windows')
+@pytest.mark.parametrize(
+    'url, expected', (
+            ('http://192.168.0.1:5000/', True),
+            ('http://192.168.0.1/', True),
+            ('http://172.16.1.1/', True),
+            ('http://172.16.1.1:5000/', True),
+            ('http://localhost.localdomain:5000/v1.0/', True),
+            ('http://172.16.1.22/', False),
+            ('http://172.16.1.22:5000/', False),
+            ('http://google.com:5000/v1.0/', False),
+    ))
+def test_should_bypass_proxies_win_registry(url, expected, monkeypatch):
+    """Tests for function should_bypass_proxies to check if proxy
+    can be bypassed or not with Windows registry settings
+    """
+    if compat.is_py3:
+        import winreg
+    else:
+        import _winreg as winreg
```

Comment:
Then we would need a platform check in compat and winreg could be `None` on non Windows platforms...